From 11d1079e75c5dd6cc8d2c5013e8caadbc52a8a2a Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Wed, 4 Feb 2015 14:19:43 +0100
Subject: [PATCH] ASoC: kirkwood: extend Armada 370 DB board driver to support
 Armada 38x DB

	Marvell Armada 38x DB platform uses very similar audio controller to the
	one in Armada 370 and also it uses same audio module supported on
	Armada 370 DB.

Change-Id: I5a13fee5c5d5522e3f0b1750bd511fcb68c7210c
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/16539
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
---
 .../bindings/sound/armada-370db-audio.txt          | 12 ++++++------
 sound/soc/kirkwood/armada-370-db.c                 | 14 +++++++++-----
 2 files changed, 15 insertions(+), 11 deletions(-)

diff --git a/Documentation/devicetree/bindings/sound/armada-370db-audio.txt b/Documentation/devicetree/bindings/sound/armada-370db-audio.txt
index bf984d238620..a7a197ac7c85 100644
--- a/Documentation/devicetree/bindings/sound/armada-370db-audio.txt
+++ b/Documentation/devicetree/bindings/sound/armada-370db-audio.txt
@@ -1,19 +1,19 @@
-Device Tree bindings for the Armada 370 DB audio
-================================================
+Device Tree bindings for the Armada 370/385 DB audio
+====================================================
 
 These Device Tree bindings are used to describe the audio complex
-found on the Armada 370 DB platform.
+found on the Armada 370 DB and Armada 385 DB platforms.
 
 Mandatory properties:
 
- * compatible: must be "marvell,a370db-audio"
+ * compatible: must be "marvell,a370db-audio" or "marvell,a385db-audio"
 
  * marvell,audio-controller: a phandle that points to the audio
-   controller of the Armada 370 SoC.
+   controller of the Armada 370/385 SoC.
 
  * marvell,audio-codec: a set of three phandles that points to:
 
-    1/ the analog audio codec connected to the Armada 370 SoC
+    1/ the analog audio codec connected to the Armada 370/A385 SoC
     2/ the S/PDIF transceiver
     3/ the S/PDIF receiver
 
diff --git a/sound/soc/kirkwood/armada-370-db.c b/sound/soc/kirkwood/armada-370-db.c
index 19b5cdd1fb15..97ac31b4141e 100644
--- a/sound/soc/kirkwood/armada-370-db.c
+++ b/sound/soc/kirkwood/armada-370-db.c
@@ -37,6 +37,9 @@ static int a370db_hw_params(struct snd_pcm_substream *substream,
 	case 96000:
 		freq = 24576000;
 		break;
+	case 192000:
+		freq = 49152000;
+		break;
 	}
 
 	return snd_soc_dai_set_sysclk(codec_dai, 0, freq, SND_SOC_CLOCK_IN);
@@ -84,7 +87,7 @@ static struct snd_soc_dai_link a370db_dai[] = {
 };
 
 static struct snd_soc_card a370db = {
-	.name = "a370db",
+	.name = "a370db/385db",
 	.owner = THIS_MODULE,
 	.dai_link = a370db_dai,
 	.num_links = ARRAY_SIZE(a370db_dai),
@@ -139,21 +139,22 @@ static int a370db_probe(struct platform_device *pdev)
 
 static const struct of_device_id a370db_dt_ids[] = {
 	{ .compatible = "marvell,a370db-audio" },
+	{ .compatible = "marvell,a385db-audio" },
 	{ },
 };
 MODULE_DEVICE_TABLE(of, a370db_dt_ids);
 
-static struct platform_driver a370db_driver = {
+static struct platform_driver a370_a385_db_driver = {
 	.driver		= {
-		.name	= "a370db-audio",
+		.name	= "a370_a385-db-audio",
 		.of_match_table = of_match_ptr(a370db_dt_ids),
 	},
 	.probe		= a370db_probe,
 };
 
-module_platform_driver(a370db_driver);
+module_platform_driver(a370_a385_db_driver);
 
 MODULE_AUTHOR("Thomas Petazzoni <thomas.petazzoni@free-electrons.com>");
-MODULE_DESCRIPTION("ALSA SoC a370db audio client");
+MODULE_DESCRIPTION("ALSA SoC a370_a385-db audio client");
 MODULE_LICENSE("GPL");
 MODULE_ALIAS("platform:a370db-audio");
-- 
